service/src/test/java/bio/terra/workspace/common/GcsBucketUtils.java

```diff
@@ -153,17 +154,36 @@ public static void assertBucketFiles(
     assertEquals(fileContent, actualContents);
   }
 
-  /** Asserts table is populated as per populateBqTable(). */
+  /**
+   * Asserts that the bucket is empty. Retries for 5 minutes to accommodate permission propagation
+   * delay
+   */
   public void assertBucketHasNoFiles(
-      AuthenticatedUserRequest userRequest, String projectId, String bucketName) {
+      AuthenticatedUserRequest userRequest, String projectId, String bucketName) throws Exception {
     StorageCow storageCow = crlService.createStorageCow(projectId, userRequest);
+    RetryUtils.getWithRetryOnException(
+        () -> countFiles(storageCow, bucketName),
+        /*totalDuration=*/ Duration.ofMinutes(5),
+        /*initialSleep=*/ Duration.ofSeconds(5),
+        /*factorIncrease=*/ 1.0,
+        /*sleepDurationMax=*/ Duration.ofSeconds(30),
+        null);
+
     int numFiles = 0;
     for (BlobCow blob : storageCow.get(bucketName).list().iterateAll()) {
       numFiles++;
     }
     assertEquals(0, numFiles);
   }
 
+  private Integer countFiles(StorageCow storageCow, String bucketName) {
+    int numFiles = 0;
+    for (BlobCow blob : storageCow.get(bucketName).list().iterateAll()) {
```

Comment:
.size()?

Reply:
That is old code I meant to delete. The `getWithRetryOnException` returns the int from the countFiles call.